c3.php

```diff
@@ -8,420 +8,463 @@
+    /**
+     * @param string $dir directory from where to attempt autoloading Codeception
+     */
+    public function setAutoloadRootDir($dir)
+    {
+        $this->autoloadRootDir = $dir;
     }
-}
 
-// phpunit codecoverage shimming
-if (!class_exists('PHP_CodeCoverage') and class_exists('SebastianBergmann\CodeCoverage\CodeCoverage')) {
-    class_alias('SebastianBergmann\CodeCoverage\CodeCoverage', 'PHP_CodeCoverage');
-    class_alias('SebastianBergmann\CodeCoverage\Report\Text', 'PHP_CodeCoverage_Report_Text');
-    class_alias('SebastianBergmann\CodeCoverage\Report\PHP', 'PHP_CodeCoverage_Report_PHP');
-    class_alias('SebastianBergmann\CodeCoverage\Report\Clover', 'PHP_CodeCoverage_Report_Clover');
-    class_alias('SebastianBergmann\CodeCoverage\Report\Crap4j', 'PHP_CodeCoverage_Report_Crap4j');
-    class_alias('SebastianBergmann\CodeCoverage\Report\Html\Facade', 'PHP_CodeCoverage_Report_HTML');
-    class_alias('SebastianBergmann\CodeCoverage\Report\Xml\Facade', 'PHP_CodeCoverage_Report_XML');
-    class_alias('SebastianBergmann\CodeCoverage\Exception', 'PHP_CodeCoverage_Exception');
-}
-// phpunit version
-if (!class_exists('PHPUnit_Runner_Version') && class_exists('PHPUnit\Runner\Version')) {
-    class_alias('PHPUnit\Runner\Version', 'PHPUnit_Runner_Version');
-}
+    /**
+     * Run Codeception C3 Coverage collection.
+     *
+     * @throws Exception\ConfigurationException
+     */
+    public function run()
+    {
+        if (isset($_COOKIE['CODECEPTION_CODECOVERAGE'])) {
+            $cookie = json_decode($_COOKIE['CODECEPTION_CODECOVERAGE'], true);
 
-// Load Codeception Config
-$configDistFile = realpath(__DIR__) . DIRECTORY_SEPARATOR . 'codeception.dist.yml';
-$configFile = realpath(__DIR__) . DIRECTORY_SEPARATOR . 'codeception.yml';
+            // fix for improperly encoded JSON in Code Coverage cookie with WebDriver.
+            // @see https://github.com/Codeception/Codeception/issues/874
+            if (!is_array($cookie)) {
+                $cookie = json_decode($cookie, true);
+            }
 
-if (isset($_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'])) {
-    $configFile = realpath(__DIR__) . DIRECTORY_SEPARATOR . $_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'];
-}
-if (file_exists($configFile)) {
-    // Use codeception.yml for configuration.
-} elseif (file_exists($configDistFile)) {
-    // Use codeception.dist.yml for configuration.
-    $configFile = $configDistFile;
-} else {
-    __c3_error(sprintf("Codeception config file '%s' not found", $configFile));
-}
-try {
-    \Codeception\Configuration::config($configFile);
-} catch (\Exception $e) {
-    __c3_error($e->getMessage());
-}
+            if ($cookie) {
+                foreach ($cookie as $key => $value) {
+                    if (!empty($value)) {
+                        $_SERVER["HTTP_X_CODECEPTION_" . strtoupper($key)] = $value;
```

Comment:
Don't use the global `$_SERVER` instead add a constructor param `$server`.
This makes the dependencies clear from the call signature:
```php

(new C3($_SERVER, $_REQUEST, $argument1, ...))->run();
```